source/extensions/common/wasm/wasm.cc

```diff
@@ -1018,7 +1018,7 @@ void Wasm::getFunctions() {
 #undef _GET
 }
 
-Wasm::Wasm(const Wasm& wasm) {
+Wasm::Wasm(const Wasm& wasm) : enable_shared_from_this<Wasm>() {
```

Comment:
What compiler do you use and what warning are you getting?

The [`Wasm` class already inherits this](https://github.com/istio/envoy/blob/9dc69fdab263ab9895c787d835b4b5131fafff04/source/extensions/common/wasm/wasm.h#L253), and both constructors are `public`, so this shouldn't be needed, AFAIK.

Reply:
Ah, sorry, I see that you've pasted the error in the opening message.

Does this change look reasonable to you, @jplevyak?

Reply:
I don't think this is necessary.  By default C++ will use the copy constructor from any public superclass.  Perhaps this is related to the issue you fixed where enabl_shared_from_this was not public?

Reply:
@jplevyak I agree with you that the behavior doesn't change with or without the commit.
It's compiler warning not an error. As this is the only low prior error at my side, I prefer to fix it.
```
[-Werror=extra]
```

Reply:
What I am trying to figure out is how you are seeing this.  Did you add -Werror=extra in your build?  I tried to rebuild that file and I am not seeing it.  Could you attach repro instructions.